src/Auth.js

```diff
@@ -429,10 +422,10 @@ const handleAuthDataValidation = async (authData, req, foundUser) => {
   );
   // Perform validation as step-by-step pipeline for better error consistency
   // and also to avoid to trigger a provider (like OTP SMS) if another one fails
-  return reducePromise(
-    // apply sort to run the pipeline each time in the same order
-    Object.keys(authData).sort(),
-    async (acc, provider) => {
+  const acc = { authData: {}, authDataResponse: {} };
+  for (const provider of Object.keys(authData).sort()) {
+    let method = '';
+    try {
       if (authData[provider] === null) {
         acc.authData[provider] = null;
         return acc;
```

Comment:
Bug found here `return acc;`

By chance i found it using this PR in my project. I'm really not a huge fan of the combination of `for` `continue` `mutating` and early return strategies. It could lead to these kinds of critical bugs, hard to catch sometimes because it's related to a functional branch and how many loops `for` will perform.

I strongly suggest reverting to my original code using the `reducePromise` strategy, for safer execution by design using the functional approach that i used. (also on `challenge` part.)

Also we need to write a new test that cover the bug that i found.

The process of the fail:
- Create auth adapter A, auth adapter B
- Create/Setup a user with adapter A
- then update the user with `{authData: { adapterA: null, adapterB: {id: "test"}}}`
- BUG: if you get the user after the update, you should notice that user has no `adapterB` in authData

Note: this bug could be flaky depending of the `provider` name. Since we sort() keys for the `for`. Using provider names like `providerA` and `providerB` should be stable to reproduce this critical bug.